### ISSUE TITLE
Heimdall chain halting after validator-join tx

### ISSUE THREAD

**Issue by ssandeep** [maticnetwork/heimdall#646]
Steps to reproduce (with single node, in local):
1. Reset heimdalld, bridge, bor. Start them again.
2. Using the setup in https://www.notion.so/maticnetwork/Setup-for-Testing-3f04a9dc2c5e44e69f258f1ec179ea7c , send the validator-join tx in heimdall:
```
$ ./build/heimdalld tx staking validator-join --tx-hash 0x227fce993d1b4d9946c30a15aa83ab26edb4a5e9670490ff32059e054facaea1 --proposer 0x6ce6134649ae82bd15a0a6f87d3efa6fc7014919 --chain-id test_heimdall --from 0x6ce6134649ae82bd15a0a6f87d3efa6fc7014919 --log-index 0
D[2021-04-18|16:26:27.281] Tx included in block                         block=19 tx=0x227fce993d1b4d9946c30a15aa83ab26edb4a5e9670490ff32059e054facaea1
D[2021-04-18|16:26:27.284] Latest block on main chain obtained          Block=9235
{"body":{"messages":[{"@type":"/heimdall.staking.v1beta1.MsgValidatorJoin","from":"0x6ce6134649ae82bd15a0a6f87d3efa6fc7014919","id":10,"activation_epoch":"0","amount":"100000000000000000000","signer_pub_key":"0x0399732839aa8cd4c206d9c3b2c686902b48f962d4b9caa37fefcaed8fdcb41534","tx_hash":"0x227fce993d1b4d9946c30a15aa83ab26edb4a5e9670490ff32059e054facaea1","log_index":"0","block_number":"19","nonce":"0"}],"memo":"","timeout_height":"0","extension_options":[],"non_critical_extension_options":[]},"auth_info":{"signer_infos":[],"fee":{"amount":[],"gas_limit":"200000","payer":"","granter":""}},"signatures":[]}

confirm transaction before signing and broadcasting [y/N]: y
{"height":"4","txhash":"21D4914593D64CECAFC164A675A7059920D8CC5BF556EE6337539155B6F6FB41","codespace":"","code":0,"data":"0A100A0E76616C696461746F722D6A6F696E","raw_log":"[{\"events\":[{\"type\":\"message\",\"attributes\":[{\"key\":\"action\",\"value\":\"validator-join\"}]},{\"type\":\"validator-join\",\"attributes\":[{\"key\":\"module\",\"value\":\"staking\"},{\"key\":\"validator-id\",\"value\":\"10\"},{\"key\":\"signer\",\"value\":\"0\"}]}]}]","logs":[{"msg_index":0,"log":"","events":[{"type":"message","attributes":[{"key":"action","value":"validator-join"}]},{"type":"validator-join","attributes":[{"key":"module","value":"staking"},{"key":"validator-id","value":"10"},{"key":"signer","value":"0"}]}]}],"info":"","gas_wanted":"200000","gas_used":"56946","tx":null,"timestamp":""}
```
3. The tx is included at height=4 and heimdall stops producing blocks beyond height=7. Heimdall logs:
```
$ ./build/heimdalld start
I[2021-04-18|16:26:15.812] starting ABCI with Tendermint                module=main
I[2021-04-18|16:26:16.268] Starting StateSync service                   module=statesync impl=StateSync
I[2021-04-18|16:26:21.469] Executed block                               module=state height=1 validTxs=0 invalidTxs=0
I[2021-04-18|16:26:21.499] Committed state                              module=state height=1 txs=0 appHash=7E9D37F67303CE87C89C854DFADD7027E36828343A73D09D5512CB313907F838
I[2021-04-18|16:26:26.585] Executed block                               module=state height=2 validTxs=0 invalidTxs=0
I[2021-04-18|16:26:26.610] Committed state                              module=state height=2 txs=0 appHash=D188542DB923367937CFA91C7760CD019EA49337F7C61AE579592F6C6D1E35AB
I[2021-04-18|16:26:31.751] Executed block                               module=state height=3 validTxs=0 invalidTxs=0
I[2021-04-18|16:26:31.779] Committed state                              module=state height=3 txs=0 appHash=0468BE2B2610669F14F146DC80AEC645EA7E742780306205FC5DEF378B3C1BA1
D[2021-04-18|16:26:36.915] Tx included in block                         block=19 tx=0x227fce993d1b4d9946c30a15aa83ab26edb4a5e9670490ff32059e054facaea1
D[2021-04-18|16:26:36.918] Latest block on main chain obtained          Block=9237
I[2021-04-18|16:26:36.919] Executed block                               module=state height=4 validTxs=1 invalidTxs=0
I[2021-04-18|16:26:36.944] Committed state                              module=state height=4 txs=1 appHash=ADAB93F1DE5BE6DDB39796C059D655F502681A6B15C8476E0B9E3DF0692A5AD2
I[2021-04-18|16:26:42.059] Executed block                               module=state height=5 validTxs=0 invalidTxs=0
I[2021-04-18|16:26:42.085] Committed state                              module=state height=5 txs=0 appHash=4C177B2ECCEF5105BA2BB5083196FF50BD6C4E0A594A0A6A042F15AC9B6138CA
D[2021-04-18|16:26:47.196] [ENDBLOCK] Updated current validator set     module=app proposer="Validator{1 0xf3d853dbaa6f5f2e42172d37eb39b9895f810541 0x03d8abb4e7c1b75ab80fd868954327b061b4f5aaccac3ff640d7b0b4659826e78c VP:1 A:-43}"
I[2021-04-18|16:26:47.196] Executed block                               module=state height=6 validTxs=0 invalidTxs=0
I[2021-04-18|16:26:47.221] Updates to validators                        module=state updates=6CE6134649AE82BD15A0A6F87D3EFA6FC7014919:100
I[2021-04-18|16:26:47.222] Committed state                              module=state height=6 txs=0 appHash=C5DBEF625078DBA3CCFB09C0DEDB5F98F8189472CEFC87208EB488B1F8B74CD1
I[2021-04-18|16:26:52.341] Executed block                               module=state height=7 validTxs=0 invalidTxs=0
I[2021-04-18|16:26:52.367] Committed state                              module=state height=7 txs=0 appHash=58BFA88CAD1ECA0E4CACF7263F19460B358397D128F63B5198C79C91A177B5A3





E[2021-04-18|16:27:33.072] error while stopping connection              module=rpc-server protocol=websocket error="already stopped"
^CI[2021-04-18|16:27:38.367] Stopping StateSync service                   module=statesync impl=StateSync
E[2021-04-18|16:27:38.367] Stopped accept routine, as transport is closed module=p2p numPeers=0
E[2021-04-18|16:27:38.367] Error serving server                         module=node err="accept tcp 127.0.0.1:26657: use of closed network connection"
```

**Comment by ssandeep** [comment]
Update: This was due to not enough voting power for the first validator. So as soon as the second validator-join 'ed, the block production stopped. So we increased the voting power of the first validator in genesis and reset the chain. Now, the blocks are producing fine even after validator-join. But the issue is there when we stop and start the chain again.

To verify, @jdkanani suggested to add two validators in the genesis file itself and we got the same behaviour.

The hypothesis now is: when we have more than one validator in the validator-set, we should have more than one validator also running. @murthyvitwit to check with @anilCSE and confirm if this is the case in cosmos.
